### PR TITLE
feat(api): access response properties from api errors

### PR DESCRIPTION
**Pull request by AllanZhengYP**
#### Description of changes:
Add guidance to access the HTTP response from REST API errors.

#### Related GitHub issue #, if available:
https://github.com/aws-amplify/amplify-js/pull/12835

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [x] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [x] JS
- [ ] Swift
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [x] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/fragments/lib/restapi/js/fetch.mdx [hunk 1]
```diff
@@ -102,3 +102,36 @@ You can not consume the response payload more than once.
...
+* `body`: HTTP response body as a string
+
+The following example shows how to access the HTTP response from an `ApiError` instance, so that you can handle the 
+error response from your REST API endpoint:
+
+```javascript
+import { ApiError } from 'aws-amplify/api';
+
+try {
+  await restOperation.response;
```
Comment (renebrandel) [inline src/fragments/lib/restapi/js/fetch.mdx:121]: maybe be a bit more explicit here with restOperation by adding a line like:

```js
const restOperation = get('/...')
```

Comment (renebrandel) [reply src/fragments/lib/restapi/js/fetch.mdx:121]: will probably also need to update the import to include "get"

**Review by renebrandel — DISMISSED** [review]

File: src/fragments/lib/restapi/js/fetch.mdx [hunk 2]
```diff
@@ -102,3 +102,40 @@ You can not consume the response payload more than once.
 
 </Callout>
 
+## Access HTTP response from errors
+
+The REST API handler may throw an `ApiError` error instance. In case the error is caused by an HTTP response with
```
Comment (jimblanc) [inline src/fragments/lib/restapi/js/fetch.mdx:107]:
```suggestion
The REST API handler may throw an `ApiError` error instance. If the error is caused by an HTTP response with a
```

**Review by jimblanc — COMMENTED** [review]
LGTM, one nit

**Review by jimblanc — APPROVED** [review]

**Review by jimblanc — APPROVED** [review]

**Review by renebrandel — APPROVED** [review]

**Pull request merged**